src/dodal/utils.py

```diff
@@ -236,6 +309,16 @@ def device_one(): and def device_two(device_one: Readable):, indicate taht
             yield name
 
 
+def get_device_factories(
```

Comment:
Should: Can we have a test around this please?

Reply:
Could: It's not 100% clear what the difference is between this and `collect_factories` and when one should use this or the other. It would be good to have some more in the docstrings or change the function names to make it clearer

Reply:
Hmm, they are very similar, maybe we should just mark `collect...` as deprecated and have done with it?

Reply:
Marking as depreciated works for me, with an explaining comment/link to issue

Reply:
I just nixxed this implementation and reworded the ADR: the only difference was that it returned a dict[str, DeviceInitilizationController] rather than a dict[str, Callable[[], AnyDevice]: as beamlines move over this can be converted to all return DIC, and when we call it from blueapi we can check if it's a DIC and handle appropriately